### PR TITLE
Navigation

### PR DESCRIPTION
**Pull request by sethnejame**
# Navigation
Pivotal Tracker Link:
https://www.pivotaltracker.com/story/show/167284957

Adding navbar feature for easy navigation.  User story below:

_As a visitor,
In order to navigate to different sections of the site,
I would like to be able to use a navbar_

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get 'main/index'
+  
+  root 'main#index'
```
Comment (oliverochman) [inline config/routes.rb]: please use the correct syntax here
you only need one of these lines

File: features/step_definitions/assertion_steps.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+  
+Then("I should see {string}") do |content|
+    expect(page).to have_content content
+  end
+
+Then("I should be on the {string} page") do |content|
```
Comment (oliverochman) [inline features/step_definitions/assertion_steps.rb]: Why do you have this step def?

File: features/step_definitions/basic_steps.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+When("I visit the site") do
+    visit root_path
+  end
+
+When("I click on {string}") do |string|
```
Comment (oliverochman) [inline features/step_definitions/basic_steps.rb]: Why do you have all of these step defs that are not used in the feature file?

File: features/navbar.feature [hunk 4]
```diff
@@ -0,0 +1,13 @@
+Feature: Visitor Can Navigate via Navbar
+    As a visitor,
+    In order to navigate to different sections of the site,
+    I would like to be able to use a navbar
+
+  Scenario: Visitor Sees Navbar Links on Main Page
+    When I visit the site
+    Then I should see "HOME"
```
Comment (oliverochman) [inline features/navbar.feature]: please make sure to refactor the steps here, everything starts with a `Given`
fix the indentation starting at line 6
`And I should see`

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -7,9 +7,25 @@
 
     <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
+    <link href='https://fonts.googleapis.com/css?family=Playfair Display' rel='stylesheet'>
   </head>
 
   <body>
+    <!-- insert navbar -->
+    <nav id='fixed-nav-bar'>
+      <div class="">
```
Comment (oliverochman) [inline app/views/layouts/application.html.erb]: why do you need this div with no `class` here?

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -7,9 +7,25 @@
...
+        <ul style="display: inline-block;">
```
Comment (oliverochman) [inline app/views/layouts/application.html.erb]: cant you extract `display: inline` & `display: inline-block;` to a css class?

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 1 above)
Comment (sethnejame) [reply config/routes.rb]: Fixed

File: features/step_definitions/assertion_steps.rb (hunk 2 above)
Comment (sethnejame) [reply features/step_definitions/assertion_steps.rb]: Originally I was going to build separate pages for each link just to make sure the links were wired correctly. . .and I made the step def for it.  But then decided that was out of scope and we just needed navigation so I trimmed it out. I'll remove this until we use it.

Fixed!

File: features/step_definitions/basic_steps.rb (hunk 3 above)
Comment (sethnejame) [reply features/step_definitions/basic_steps.rb]: These extra step defs were brought in from an old project.  I can remove them.

File: features/navbar.feature (hunk 4 above)
Comment (sethnejame) [reply features/navbar.feature]: Fixed

File: app/views/layouts/application.html.erb (hunk 5 above)
Comment (sethnejame) [reply app/views/layouts/application.html.erb]: These will be styled later

File: app/views/layouts/application.html.erb (hunk 6 above)
Comment (sethnejame) [reply app/views/layouts/application.html.erb]: Done

File: config/routes.rb (hunk 1 above)
Comment (oliverochman) [reply config/routes.rb]: please refactor to
`root controller: :main, action: :index`

File: features/step_definitions/basic_steps.rb (hunk 3 above)
Comment (oliverochman) [reply features/step_definitions/basic_steps.rb]: You don't click on anything in this feature so please remove the step on line 5

File: features/navbar.feature [hunk 7]
```diff
@@ -0,0 +1,13 @@
+Feature: Visitor Can Navigate via Navbar
+    As a visitor,
+    In order to navigate to different sections of the site,
+    I would like to be able to use a navbar
+
+  Scenario: Visitor Sees Navbar Links on Main Page
```
Comment (oliverochman) [inline features/navbar.feature:6]: This line should be indented the same as line 4
And then fix the indentation underneath when u have done that

File: features/navbar.feature [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+      Given I visit the site
+      Then I should see "HOME"
+      Then I should see "LOGIN"
```
Comment (oliverochman) [inline features/navbar.feature]: Please refactor to 
`And I should see 'Login'`

etc etc

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: features/step_definitions/basic_steps.rb (hunk 3 above)
Comment (sethnejame) [reply features/step_definitions/basic_steps.rb]: Done!

File: config/routes.rb (hunk 1 above)
Comment (sethnejame) [reply config/routes.rb]: Done!

File: features/navbar.feature (hunk 8 above)
Comment (sethnejame) [reply features/navbar.feature]: Done

**Review by oliverochman — APPROVED** [review]

**Pull request merged**